### DISCUSSION TITLE
Is it possible to create a virtual com port in WSL and expose it in windows?

### DISCUSSION THREAD

**Discussion by joezappie** [microsoft/WSL#9289] [Q&A]
I'd like to use socat to make a virtual serial com port in WSL and then be able to access it via the Windows host like a normal com port. Where it shows up in the Device Manager as a COM port - Is this possible?

My reason for this is I'm using a VSCode devcontainer to run an application that needs RS232 communication. This normally runs on embedded hardware, but for development I want to run it locally in a container, simulating what I can. I then have an application in windows for simulating my serial data. I need to connect that application, to my application running in the devcontainer. 

Right now, I'm getting around this by using 2 USB serial adapters and passing one of them to WSL using usbipd. My windows application connects to one, while the other I pass to my devcontainer. While this works, it would be way more awesome to do it all virtually and not need any hardware. 

In this diagram, I'm showing Putty, but really I'd be using a custom application that connects to a serial device.
![Untitled Diagram drawio (3)](https://user-images.githubusercontent.com/6005836/206634199-62a44993-cd40-4124-b7a4-8f95dcbb23c1.png)

TLDR: Trying to do the reverse of usbpid - Pass linux com port to windows, not windows to linux.

Any ideas?
